### ISSUE TITLE
Items with `tabindex="0"` and no role are silent in JAWS 2021 with FireFox

### ISSUE THREAD

**Issue by rskrzypczynski** [FreedomScientific/VFO-standards-support#519]
### Summary

Items with `tabindex="0"` and no role are silent in JAWS 2021 with FireFox

### Expected result

> Items with `tabindex="0"` should be announced by screen reader

### Actual result

> Items with `tabindex="0"` and no role are silent in JAWS 2021 but works as expected in 2020, and the issue is only for FF (Versions 83 and 88), other browsers are OK

### Example

1. Go to https://jsfiddle.net/ojefy8xq/1/

### Additional Information

Element with `tabindex="0"` and `role="gridcell"` works properly

#### JAWS version and build number

2021.2103.174
 
#### Operating System and version

Windows 10 64bit

#### Browser and version:

Tested on FireFox 88 & 83

**Comment by hrvojegolcic** [comment]
I can also confirm there is a difference between JAWS 2020 and 2021. The new version is silent in FF when I TAB into the element. It only work if it has `role=button` or, as you said `role="gridcell"`

**Comment by JAWS-test** [comment]
This appears to be a Firefox bug and not one from JAWS. The span element with tabindex does not have a name in Firefox. It may not be a Firefox bug because the Accname specification does not provide a name for span elements (see https://github.com/w3c/accname/issues/120). Chrome behaves the same way when the span has no tabindex. Only with tabindex Chrome considers the element relevant and assigns a name.

**Comment by JAWS-test** [comment]
The question would be why a span should get the focus. In web pages only interactive elements should get the focus. And interactive elements need a role. Once the element has a role, the name should also be output correctly

**Comment by hrvojegolcic** [comment]
There is definitely something off with JAWS 2021 specifically, because **with JAWS 2020 the issue does not occur** in the very same browser. For that matter it also does not occur with other browsers (or even other screen readers).
So clearly the specific combination of FF+JAWS 2021 is broken.

Going further into the topic I have made another [jsfiddle example](https://jsfiddle.net/8hv0tc2j/3/) as following;

* TABINDEX
* TABINDEX + IMG ROLE + ARIA LABEL
* TABINDEX + ROLE NONE + ARIA LABEL
* TABINDEX + BUTTON
* TABINDEX + GRIDCELL

In the example even the elements that have DO HAVE the "name" in Firefox, such as those with `aria-label`, are still silent when focused in JAWS 2021, unlike JAWS 2020 where it works for all.

About interactive elements not having a `role` imo is not the point of discussion here. While it can be discussed it's wrong surely there are many occurrences across the Web where `tabindex` is used, which are now inaccessible since JAWS 2021. Personally I could think of cases where one uses `tabindex` but does not want to use a `button` or `gridcell` role, clearly because the item is not a button, and so far there is yet no other appropriate role. If this is not fixed I see developers then using `gridcell` role, as a workaround,  which is even worse. To summarize, when item is made focusable by setting `tabindex` surely the JAWS should not be silent when navigated there, especially if it works in other browsers.

**Comment by JAWS-test** [comment]
@hrvojegolcic Thank you for the second example. Now I'm completely convinced it's a JAWS bug, because the output doesn't happen even when Firefox submits a name to the Accessibility API. This was not the case with the first example, so I was leaning towards it being fixed by Firefox, even if it worked in JAWS 2020, because screen readers should be able to rely on the information in the API.

**Comment by JAWS-test** [comment]
@rskrzypczynski Maybe you can delete the "no role" in the issue title, because the problem also occurs with elements that have a role (e.g. graphics).

**Comment by JAWS-test** [comment]
Among others, the following HTML elements and ARIA roles have no output at all in Firefox + JAWS 2021 when navigating with the tab key:

- HTML: div, span, p, li, strong, sub, table, a (without href), img (with and without alt)
- ARIA: alert, application, banner, blockquote, cell, code, definition, deletion, document, form, generic, grid, group (only without label), img  (only without label), log, math, meter, none, paragraph, presentation, radiogroup  (only without label), region, row, 

scrollbar, search, status, strong, subscript, tablist (only without label), tabpanel, table, time, timer, tooltip

**Comment by juliemoynatPro** [comment]
I have the same problem with `p` or `div` elements with a `tabindex="-1"` which receive the focus via JavaScript.
The problem is observed with JAWS 2021 and 2022 + Firefox or Chrome. There is no problem with JAWS 2020 and it's fixed in JAWS 2023.
